src/metabase/query_processor/util/persisted_cache.clj

```diff
@@ -0,0 +1,50 @@
+(ns metabase.query-processor.util.persisted-cache
+  (:require    [clojure.string :as str]
+               [metabase.driver :as driver]
+               [metabase.driver.ddl.interface :as ddl.i]
+               [metabase.driver.sql.util :as sql.u]
+               [metabase.driver.util :as driver.u]
+               [metabase.models.persisted-info :as persisted-info]
+               [metabase.public-settings :as public-settings]))
+
+(defn- segmented-user?
+  []
+  (if-let [segmented? (resolve 'metabase-enterprise.sandbox.api.util/segmented-user?)]
```

Comment:
Double-check:
The function `segmented-user?` could throw an exception in case no `current-permissions` are found.

Does this safe when we run it in the context of a background task?

Reply:
It's a good call out. I would prefer a function that doesn't throw but i'm nervous to catch the exceptions because in this case they feel truly exceptional.

I don't believe we'll bind a user for background tasks. But I can't think of any scenario where we would hit the persistence in a background task. I'll ask @camsaul for some thoughts.

Reply:
Also, this change happened previously and here I just pull it into a shared util namespace. I'm gonna merge with it in this state and chat with cam about it.

Reply:
I like your version. having the `persists-fn` is convienet for cases where we want to test things before the persistent happens.